demo_usage/convert_tracking_dataset_to_forecasting.py

```diff
@@ -0,0 +1,196 @@
+	box_list = []
+	for label in sweep_labels:
+
+		x = label['center']['x']
+		y = label['center']['y']
+		z = label['center']['z']
+
+		qw = label['rotation']['w']
+		qx = label['rotation']['x']
+		qy = label['rotation']['y']
+		qz = label['rotation']['z']
+
+		obj_label_record = json_label_dict_to_obj_record(label)
+		vertices_egofr = obj_label_record.as_2d_bbox()
+
+		box = Box3d(
+			center = [x, y, z], # Argoverse and nuScenes use scalar-first
+			width = label['width'],
+			length = label['length'],
+			height = label['height'],
+			orientation_q = [qw, qx, qy, qz],
+			name = label['label_class'],
+			track_id = label['track_label_uuid'],
+			timestamp = timestamp
+		)
+
+		# transform box from the egovehicle frame into the city frame
+		box.center = city_SE3_egovehicle.transform_point_cloud(np.array(box.center).reshape(1,3)).squeeze()
+		box.center = list(box.center)
+		box.orientation_q = list(rotmat2quat(city_SE3_egovehicle.rotation @ quat2rotmat(list(box.orientation_q))))
+		vertices_cityfr = city_SE3_egovehicle.transform_point_cloud(vertices_egofr)
+		box.v0 = list(vertices_cityfr[0])
+		box.v1 = list(vertices_cityfr[1])
+		box.v2 = list(vertices_cityfr[2])
+		box.v3 = list(vertices_cityfr[3])
+		box.orientation_yaw = quaternion_yaw_scipy(box.orientation_q)
+		box_list.append(box)
+
+	return box_list
+
+
+def create_tracks_csv(root_path: str, split_subdir: str) -> None:
+	""" """
+	split_root_path = f'{root_path}/{split_subdir}'
+	dl = SimpleArgoverseTrackingDataLoader(data_dir=split_root_path, labels_dir=split_root_path)
+	valid_log_ids = dl.sdb.get_valid_logs()
+	# loop through all of the logs
+	for log_id in valid_log_ids:
+
+		log_row_dictionaries = []
+		# for each log, loop through all of the LiDAR sweeps
+		log_ply_fpaths = dl.get_ordered_log_ply_fpaths(log_id)
+		num_log_sweeps = len(log_ply_fpaths)
+		for sample_idx, sample_ply_fpath in enumerate(log_ply_fpaths):
+			if sample_idx % 100 == 0:
+				print(f'\t{log_id}: On {sample_idx}/{num_log_sweeps}')
+
+			sample_lidar_timestamp = int(Path(sample_ply_fpath).stem.split('_')[-1])
+			city_SE3_egovehicle = dl.get_city_SE3_egovehicle(log_id, sample_lidar_timestamp)
+			sweep_labels = dl.get_labels_at_lidar_timestamp(log_id, sample_lidar_timestamp)
+			boxes = construct_argoverse_boxes_lidarfr(sample_lidar_timestamp, sweep_labels, city_SE3_egovehicle)
+
+			log_row_dictionaries.extend([box.__dict__ for box in boxes])
+
+		csv_fpath = f'{split_subdir}_{log_id}.csv'
+		write_csv(csv_fpath, log_row_dictionaries)
+
+def write_csv(csv_fpath: str, dict_list: List[Dict[str,Any]], delimiter='\t'):
+	""" """
+	with open(csv_fpath, 'w', newline='') as csvfile:
+
+		fieldnames = dict_list[0].keys()
+		writer = csv.DictWriter(csvfile, fieldnames=fieldnames, delimiter=delimiter)
+		writer.writeheader()
+		for row_dict in dict_list:
+			writer.writerow(row_dict)
+
+
+if __name__ == '__main__':
+
+	root_path = '/home/ubuntu/argoverse/argoverse-tracking/val-1.1/argoverse-tracking'
```

Comment:
[Nit] I'm guessing the root path will need to be changed by the end user if we don't provide a CLI arg to do so. Maybe we should call out all the places the end user might need to change variables with all caps comments.